framework/MCMC/MCMC.py

```diff
@@ -0,0 +1,293 @@
+    self._updateValues = copy.copy(self._initialValues)
+
+  def initialize(self, externalSeeding=None, solutionExport=None):
+    """
+      This function should be called every time a clean MCMC is needed. Called before takeAstep in <Step>
+      @ In, externalSeeding, int, optional, external seed
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    # TODO: currently, we only consider uncorrelated case
+    # initialize distributions
+    for _, dist in self._availProposal.items():
+      dist.initializeDistribution()
+    self._acceptDist.initializeDistribution()
+    for var, dist in self._proposal.items():
+      if dist:
+        self._proposal[var] = self.retrieveObjectFromAssemblerDict('proposal', dist)
+      else:
+        self._proposal[var] = self._availProposal['normal']
+    AdaptiveSampler.initialize(self, externalSeeding=externalSeeding, solutionExport=solutionExport)
+    ## TODO: currently AdaptiveSampler is still using self.assemblerDict to retrieve the target evaluation.
+    # We should change it to using the following method.
+    # retrieve target evaluation
+    # self._targetEvaluation = self.retrieveObjectFromAssemblerDict('TargetEvaluation', self._targetEvaluation)
+
+    # initialize the input variable values
+    for key, value in self._updateValues.items():
+      totDim = self.variables2distributionsMapping[key]['totDim']
+      dist = self.distDict[key]
+      if totDim != 1:
+        self.raiseAnError(IOError,"Total dimension for given distribution {} should be 1".format(dist.type))
+      if value is None:
+        value = dist.rvs()
+        self._updateValues[key] = value
+
+  def localGenerateInput(self, model, myInput):
+    """
+      Provides the next sample to take.
+      After this method is called, the self.inputInfo should be ready to be sent
+      to the model
+      @ In, model, model instance, an instance of a model
+      @ In, myInput, list, a list of the original needed inputs for the model (e.g. list of files, etc.)
+      @ Out, None
+    """
+    for key, value in self._updateValues.items():
+      self.values[key] = value
+      self.inputInfo['SampledVarsPb'][key] = self.distDict[key].pdf(value)
+      self.inputInfo['ProbabilityWeight-' + key] = 1.
+    self.inputInfo['PointProbability'] = 1.0
+    self.inputInfo['ProbabilityWeight' ] = 1.0
+
+  def localFinalizeActualSampling(self, jobObject, model, myInput):
+    """
+      General function (available to all samplers) that finalize the sampling
+      calculation just ended. In this case, The function is aimed to check if
+      all the batch calculations have been performed
+      @ In, jobObject, instance, an instance of a JobHandler
+      @ In, model, model instance, it is the instance of a RAVEN model
+      @ In, myInput, list, the generating input
+      @ Out, None
+    """
+    AdaptiveSampler.localFinalizeActualSampling(self, jobObject, model, myInput)
```

Comment:
this method is not needed if you just call the baseclass method...you can just remove it